plugins/modules/dw_vw.py

```diff
@@ -0,0 +1,440 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cloudera.cloud.plugins.module_utils.cdp_common import CdpModule
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: dw_vw
+short_description: Create CDP Data Warehouse Virtual Warehouse
+description:
+    - Create CDP Virtual Warehouse
+author:
+  - "Webster Mudge (@wmudge)"
+  - "Dan Chaffelson (@chaffelson)"
+  - "Saravanan Raju (@raju-saravanan)"
+requirements:
+  - cdpy
+options:
+  id:
+    description:
+      - If an ID is provided, that Virtual Warehouse will be deleted if C(state=absent)
+    type: str
+    required: When state is absent
+  cluster_id:
+    description: ID of cluster where Virtual Warehouse should be created.
+    type: str
+    required: True
+  dbc_id:
+    description: ID of Database Catalog that the Virtual Warehouse should be attached to.
+    type: str
+    required: True
+  vw_type:
+    description: Type of Virtual Warehouse to be created.
+    type: str
+    required: True
+  name:
+    description: Name of the Virtual Warehouse.
+    type: str
+    required: True
+  template:
+    description: Name of configuration template to use.
+    type: str
+    required: False
+  autoscaling_min_cluster:
+    description: Minimum number of available nodes for Virtual Warehouse autoscaling.
+    type: int
+    required: False
+  autoscaling_max_cluster:
+    description: Maximum number of available nodes for Virtual Warehouse autoscaling.
+    type: int
+    required: False
+  config:
```

Comment:
There is no `config` parameter group in the code -- `commonConfigs`, `applicationConfigs`, `ldapGroups`, and `enableSSO` are all "top level" module parameters in the code. Moreover, the parameters are lowercase with underscores.

Reply:
My bad, will fix this.